src/service/resource.js

```diff
@@ -357,10 +365,18 @@ export class Resource {
     const viewport = this.resources_.getViewport();
     this.layoutBox_ = box;
 
+    const win = this.resources_.win;
+    const computedStyle = win./*OK*/getComputedStyle(this.element);
```

Comment:
I don't think element.style is sufficient; it would ignore anything that comes from the CSS style rules.

I've changed it so that the getLayoutMargins method calls getComputedStyle and reads the margins only when it's called and have also changed mutate() in resources-impl.js to only call getLayoutMargins() when there is actually a newMargins defined in the request. Does that seem OK?